test/framework/machinepool_helpers.go

```diff
@@ -286,16 +287,19 @@ func getMachinePoolInstanceVersions(ctx context.Context, input GetMachinesPoolIn
 	versions := make([]string, len(instances))
 	for i, instance := range instances {
 		node := &corev1.Node{}
-		err := wait.PollImmediate(retryableOperationInterval, retryableOperationTimeout, func() (bool, error) {
-			err := input.WorkloadClusterGetter.Get(ctx, client.ObjectKey{Name: instance.Name}, node)
-			if err != nil {
+		var nodeGetError error
+		err := wait.PollImmediate(100*time.Millisecond, 10*time.Second, func() (bool, error) {
+			nodeGetError = input.WorkloadClusterGetter.Get(ctx, client.ObjectKey{Name: instance.Name}, node)
+			if nodeGetError != nil {
 				return false, nil //nolint:nilerr
 			}
 			return true, nil
 		})
 		if err != nil {
-			// Dump the instance name and error here so that we can log it as part of the version array later on.
-			versions[i] = fmt.Sprintf("%s error: %s", instance.Name, err)
+			if node != nil {
```

Comment:
Does this work as expected?

Looks like if node is  nil the error is not surfaced at all

Reply:
Also not sure why we're checking for node and not nodeGetError

Reply:
You're right - should be fixed now.